packages/grid/x-grid/src/XGrid.test.tsx

```diff
@@ -220,6 +220,47 @@ describe('<XGrid />', () => {
       fireEvent.click(header);
       expect(getColumnValues()).to.deep.equal(['Puma', 'Nike', 'Adidas']);
     });
+
+    it('should reset the sortedRows state when rows prop change', async () => {
+      interface TestCaseProps {
+        rows: any[];
+      }
+      let apiRef;
+      const TestCase = (props: TestCaseProps) => {
+        const { rows } = props;
+        apiRef = useApiRef();
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <XGrid {...defaultProps} rows={rows} apiRef={apiRef} />
```

Comment:
I think that we should include the sort model to make sure it works:

```suggestion
            <XGrid {...defaultProps} rows={rows} apiRef={apiRef} sortModel={[
                {
                  field: 'brand',
                  sort: 'asc',
                },
              ]} />
```